src/group_theory/free_abelian_group_finsupp.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import group_theory.free_abelian_group
+import data.finsupp.basic
+
+/-!
+# Isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+
+In this file we construct the canonical isomorphism between `free_abelian_group X` and `X →₀ ℤ`.
+We use this to transport the notion of `support` from `finsupp` to `free_abelian_group`.
+
+## Main declarations
+
+- `free_abelian_group.equiv_finsupp`: group isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+- `free_abelian_group.coeff`: the multiplicity of `x : X` in `a : free_abelian_group X`
+- `free_abelian_group.support`: the finset of `x : X` that occur in `a : free_abelian_group X`
+
+-/
+
+noncomputable theory
+
+open_locale big_operators
+
+variables (X : Type*)
+
+/-- The group homomorphism `free_abelian_group X →+ (X →₀ ℤ)`. -/
+def free_abelian_group.to_finsupp : free_abelian_group X →+ (X →₀ ℤ) :=
+free_abelian_group.lift $ λ x, finsupp.single x (1 : ℤ)
+
+/-- The group homomorphism `(X →₀ ℤ) →+ free_abelian_group X`. -/
+def finsupp.to_free_abelian_group : (X →₀ ℤ) →+ free_abelian_group X :=
+finsupp.lift_add_hom $ λ x, (smul_add_hom ℤ (free_abelian_group X)).flip (free_abelian_group.of x)
+
+open finsupp free_abelian_group
+
+@[simp] lemma finsupp.to_free_abelian_group_comp_single_add_hom (x : X) :
+  (finsupp.to_free_abelian_group X).comp (finsupp.single_add_hom x) =
+    (smul_add_hom ℤ (free_abelian_group X)).flip (of x) :=
+begin
+  ext,
+  simp only [add_monoid_hom.coe_comp, finsupp.single_add_hom_apply, function.comp_app,
+    one_smul, to_free_abelian_group, finsupp.lift_add_hom_apply_single]
```

Comment:
```suggestion
  one_smul, to_free_abelian_group, finsupp.lift_add_hom_apply_single],
```

Reply:
I think the indentation that Johan had is the correct one, because this line is really a continuation of the previous line.
The comma is optional (and I personally prefer to omit the comma at the end of the tactic block).

Reply:
I see, I personally prefer it but at any rate I was suggesting to keep it uniform along the whole file. But of course it is a very minor comment, I agree that this `PR` should go through.